backend/src/main/java/com/karankumar/bookproject/backend/service/UserService.java

```diff
@@ -89,6 +92,13 @@ public User register(@NonNull User user) throws UserAlreadyRegisteredException {
         Role userRole = roleRepository.findByRole(UserRole.USER.toString())
                                       .orElseThrow(() -> new AuthenticationServiceException(
                                               "The default user role could not be found"));
+
+        Strength passwordStrength = zxcvbn.measure(user.getPassword());
+        if (passwordStrength.getScore() <= 2) {
```

Comment:
User.java:

```java
    @PasswordStrengthCheck(PasswordStrength.STRONG)
    private String password;
```

UserService.java:
```java
        Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
        Set<ConstraintViolation<User>> constraintViolations = validator.validate(user);
        if (!constraintViolations.isEmpty()) {
            throw new ConstraintViolationException(constraintViolations);
        }
```

We won't need this code where you explicitly measure the password strength and check if it is sufficiently strong as we have the above code which will throw a constraint violation exception if the password is too weak.